app/modules/listings/components/listing_elements.svelte

```diff
@@ -11,44 +11,41 @@
   import { addEntitiesImages } from '#entities/lib/types/work_alt'
   import { getViewportHeight } from '#lib/screen'
 
-  export let elements, listingId, isEditable
+  export let elements = [], listingId, isEditable
 
   let flash, inputValue = '', showSuggestions
-  let entities = []
 
+  let paginatedElements = []
   const paginationSize = 15
-  let offset = paginationSize
+  let offset = 0
   let fetching
   let windowScrollY = 0
   let listingBottomEl
 
-  const getElementsEntities = async elements => {
+  const assignEntitiesToElements = async elements => {
     const uris = pluck(elements, 'uri')
     const res = await getEntitiesAttributesByUris({
       uris,
       attributes: [ 'info', 'labels', 'descriptions', 'image' ],
       lang: app.user.lang
     })
-    const serializedEntities = Object.values(res.entities).map(serializeEntity)
-    await addEntitiesImages(serializedEntities)
-    return serializedEntities
+    const entities = Object.values(res.entities).map(serializeEntity)
+    await addEntitiesImages(entities)
+    entities.forEach(assignEntityToElement(elements))
```

Comment:
[non-blocking]
We already have entities indexed by uris available
```suggestion
    const entitiesByUris = res.entities
    const entities = Object.values(entitiesByUris).map(serializeEntity)
    await addEntitiesImages(entities)
    for (const element of elements) {
      element.entity = entitiesByUris[element.uri]
    }
```
=> n iterations instead of, if I'm not mistaken, n*(n/2)

Reply:
done in 215bd197802c89439f54fe5dcf4e3af6b464f556